iothub_client/tests/common_e2e/iothubclient_common_e2e.c

```diff
@@ -144,29 +162,37 @@ static void sendCompleteCallback(void* context, IOTHUB_MESSAGING_RESULT messagin
 
 static int IoTHubCallback(void* context, const char* data, size_t size)
 {
-    (void)size;
-    int result = 0; // 0 means "keep processing"
-    EXPECTED_SEND_DATA* expectedData = (EXPECTED_SEND_DATA*)context;
-    if (expectedData != NULL)
+    int messagesReceived = 0;
+    SEND_DATA_BUNDLE* d2cMessages = (SEND_DATA_BUNDLE*)context;
+
+    for (int i = 0; i < d2cMessages->count; i++)
     {
-        if (Lock(expectedData->lock) != LOCK_OK)
-        {
-            ASSERT_FAIL("unable to lock");
-        }
-        else
+        EXPECTED_SEND_DATA* expectedData = d2cMessages->items[i];
+        if (expectedData != NULL)
         {
-            if (
-                (strlen(expectedData->expectedString) == size) &&
-                (memcmp(expectedData->expectedString, data, size) == 0)
-                )
+            if (Lock(expectedData->lock) != LOCK_OK)
             {
-                expectedData->wasFound = true;
-                result = 1;
+                ASSERT_FAIL("unable to lock");
+            }
+            else
+            {
+                if (
+                    (strlen(expectedData->expectedString) == size) &&
+                    (memcmp(expectedData->expectedString, data, size) == 0)
+                    )
+                {
+                    expectedData->wasFound = true;
```

Comment:
A log message(s) don't match could help debugging.

Reply:
Also the size of the message (given its 64KB don't think you should print whole thing out, but the expected and actual sizes can help.)